src/test/java/org/sagebionetworks/bridge/spring/controllers/MasterSchedulerControllerTest.java

```diff
@@ -0,0 +1,242 @@
+    
+    MasterSchedulerConfig mockConfig;
+    
+    UserSession session; 
+    
+    Study study;
+    
+    @BeforeMethod
+    private void before() {
+        MockitoAnnotations.initMocks(this);
+        
+        mockConfig = TestUtils.getMasterSchedulerConfig();
+        
+        study = new DynamoStudy();
+        study.setUserProfileAttributes(Sets.newHashSet("foo", "baz"));
+        study.setIdentifier(TEST_STUDY_IDENTIFIER);
+        
+        session = new UserSession();
+        session.setStudyIdentifier(TEST_STUDY);
+        session.setAuthenticated(true);
+        session.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(ADMIN)).build());
+        
+        doReturn(session).when(controller).getSessionIfItExists();
+        when(mockStudyService.getStudy(TEST_STUDY)).thenReturn(study);
+        when(mockStudyService.getStudy(TEST_STUDY_IDENTIFIER)).thenReturn(study);
+        
+        doReturn(mockRequest).when(controller).request();
+        doReturn(mockResponse).when(controller).response();
+    }
+    
+    @Test
+    public void verifyAnnotations() throws Exception {
+        assertCrossOrigin(MasterSchedulerController.class);
+        assertGet(MasterSchedulerController.class, "getAllSchedulerConfigs");
+        assertCreate(MasterSchedulerController.class, "createSchedulerConfig");
+        assertGet(MasterSchedulerController.class, "getSchedulerConfig");
+        assertPost(MasterSchedulerController.class, "updateSchedulerConfig");
+        assertDelete(MasterSchedulerController.class, "deleteSchedulerConfig");
+        assertGet(MasterSchedulerController.class, "getSchedulerStatus");
+    }
+    
+    @Test
+    public void getAllSchedulerConfigs() throws Exception {
+        session.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(ADMIN)).build());
+        List<MasterSchedulerConfig> configs = ImmutableList.of(mockConfig, mockConfig, mockConfig);
+        
+        // Mock service.
+        when(mockSchedulerService.getAllSchedulerConfigs()).thenReturn(configs);
+        
+        // Execute and validate.
+        ResourceList<MasterSchedulerConfig> result = controller.getAllSchedulerConfigs();
+        
+        assertEquals(result.getItems().size(), 3);
+        verify(mockSchedulerService).getAllSchedulerConfigs();
+    }
+    
+    @Test
+    public void createSchedulerConfig() throws Exception {
+        session.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(ADMIN)).build());
+        doReturn(session).when(controller).getAuthenticatedSession(ADMIN);
+        
+        mockRequestBody(mockRequest, mockConfig);
+        
+        // Mock service.
+        when(mockSchedulerService.createSchedulerConfig(any())).thenReturn(mockConfig);
+        
+        // Execute and validate.
+        MasterSchedulerConfig result = controller.createSchedulerConfig();
+        
+        assertEquals(result.getScheduleId(), SCHEDULE_ID);
```

Comment:
Similarly, we should verify the config returned by the controller is the same as the config that was returned by the service. (Same for update().)